inspector/src/components/graph/canvasGraphService.ts

```diff
@@ -38,17 +39,222 @@ export class CanvasGraphService {
         // First we clear the canvas so we can draw our data!
         this.clear();
 
-        // TODO: Draw each dataset
-        this.datasets.forEach((dataset: IPerfDataset) => {
-            // do drawing
+        // Get global min max of time axis (across all datasets).
+        let globalTimeMinMax = {min: Infinity, max: 0};
+
+        // TODO: Make better sliding window code (accounting for zoom and pan).
+        // Keep only visible and non empty datasets and get a certain window of items.
+        const datasets = this.datasets.filter((dataset: IPerfDataset) => !dataset.hidden && dataset.data.length > 0).map((dataset: IPerfDataset) => ({
+            ...dataset,
+            data: dataset.data.slice(Math.max(dataset.data.length - 300, 0))
+        }));
+
+        datasets.forEach((dataset: IPerfDataset) => {
+            const timeMinMax = this._getMinMax(dataset.data.map((point: IPerfPoint) => point.timestamp));            
+            globalTimeMinMax = {min: Math.min(timeMinMax.min, globalTimeMinMax.min), max: Math.max(timeMinMax.max, globalTimeMinMax.max)};

```

Comment:
instead of reassigning, make this a `const` and set min and max of the object

Reply:
Done!